src/Bls12381G2KeyPair.ts

```diff
@@ -74,7 +75,31 @@ const signerFactory = (key: Bls12381G2KeyPair): KeyPairSigner => {
     };
   }
   return {
-    async sign({ data }): Promise<Uint8Array> {
+    async sign({ data, proverCommitment }): Promise<Uint8Array> {
```

Comment:
IMHO, we explicitly define `blindSign({ data, proverCommitment })` with the original `sign({ data })`, instead of integrating into the single `sign({ data, proverCommitment})` function. While both the output type of `blindSign` and `sign` are the same (`Promise<Uint8Array>`), the former actually is a serialized `BlindSignature` (needs unblinding before verification) but the latter is a serialized `Signature`. Unifying two functions seems error-prone to me.